File: c++/arrays/1d/secondlargest.cpp
//find the second largest element in the array
#include <bits\stdc++.h>
using namespace std;

int second_largest(vector<int> &v);

int main(void) {
    int n;
    cout<<"Enter number of elements in vector: ";
    cin>>n;

    vector<int> v;
    cout<<"Enter elements in vector: ";

    for (int i=0; i<n; i++) {
        int input;
        cin>>input;
        v.emplace_back(input);
    }

    int a = second_largest(v);
    cout<<"second largest element in this arrray is: "<<a;
    return 0;
}

int second_largest(vector<int> &v) {
    int largest = v[0];
    int s_largest = INT_MIN;
    for (int i=0; i<v.size(); i++) {
        if (v[i]>largest) {
            s_largest = largest;
            largest = v[i];
        } else if (v[i]<largest && v[i]>s_largest) {
            s_largest = v[i];
        }
    }
    return s_largest;
}

File: java_dsa/arrays/second_largest.java
//find the second largest number in an array
import java.util.Scanner;
import java.lang.Integer;


public class second_largest 
{
    public static void main(String[] args) 
    {
        System.out.println("Enter number of elements in array:");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements:");
        input_arr(arr, n);
        int s = s_largest(arr);
        if (s == Integer.MIN_VALUE) 
        {
            System.out.println("all elements are same");
        } else
        {
            System.out.print("The second largest number is: "+s);
        }
        sc.close();
    }
    public static void input_arr(int[] arr,int n) 
    {
        Scanner sc = new Scanner(System.in);
        for (int i=0; i<n; i++) 
        {
            int input = sc.nextInt();
            arr[i] = input;
        }
        sc.close();
    }
    public static int s_largest(int arr[]) 
    {
        int largest = arr[0];
        int s_largest = Integer.MIN_VALUE;
        for (int i=1; i<arr.length; i++) 
        {
            if (arr[i]>largest) 
            {
                int temp = largest;
                largest = arr[i];
                s_largest = temp;
            } else if(arr[i]>s_largest && arr[i]!=largest) {
                s_largest = arr[i];
            }
        }
        return s_largest;
    }
}
